rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.retrievers/result-diversification/10_mmr_result_diversification_retriever.yml

```diff
@@ -0,0 +1,486 @@
+          {"index":{}}
+          {"textbody": "fifth text other", "textvector": [0.1, 0.9, 0.5, 0.9], "keywordfield": "test5"}
+          {"index":{}}
+          {"textbody": "sixth text other", "textvector": [0.05, 0.05, 0.05, 0.05], "keywordfield": "test6"}
+
+  - do:
+      headers:
+        Content-Type: application/json
+      bulk:
+        index: test-result-byte-diversification-index
+        refresh: true
+        body: |
+          {"index":{}}
+          {"textbody": "first text", "textvector": "0BB8", "keywordfield": "test1"}
+          {"index":{}}
+          {"textbody": "second text", "textvector": "94A2", "keywordfield": "test2"}
+          {"index":{}}
+          {"textbody": "second text duplicate", "textvector": "94A2", "keywordfield": "test2_dup"}
+          {"index":{}}
+          {"textbody": "third text", "textvector": "0560", "keywordfield": "test3"}
+          {"index":{}}
+          {"textbody": "third text duplicate", "textvector": "0560", "keywordfield": "test3_dup"}
+          {"index":{}}
+          {"textbody": "fourth text", "textvector": "2112", "keywordfield": "test4"}
+          {"index":{}}
+          {"textbody": "fifth text", "textvector": "5051", "keywordfield": "test5"}
+
+---
+teardown:
+  - do:
+      indices.delete:
+        index: test-result-diversification-index
+        ignore: 404
+
+  - do:
+      indices.refresh: { }
+
+---
+"Test MMR result diversification single index float type":
+  - do:
+      search:
+        index: test-result-diversification-index
+        body:
+          retriever:
+            knn:
+              field: "textvector"
+              query_vector: [ 0.5, 0.2, 0.4, 0.4 ]
+              k: 10
+              num_candidates: 10
+
+  - match: { hits.total.value: 8 }
+  - match: { hits.hits.0._source.textbody: "second text" }
+  - match: { hits.hits.1._source.textbody: "second text duplicate" }
+  - match: { hits.hits.2._source.textbody: "third text" }
+  - match: { hits.hits.3._source.textbody: "third text duplicate" }
+  - match: { hits.hits.4._source.textbody: "first text" }
+  - match: { hits.hits.5._source.textbody: "sixth text" }
+  - match: { hits.hits.6._source.textbody: "fourth text" }
+  - match: { hits.hits.7._source.textbody: "fifth text" }
+
+  - do:
+      search:
+        index: test-result-diversification-index
+        body:
+          retriever:
+            diversify:
+              type: "mmr"
+              field: "textvector"
+              lambda: 0.3
+              rank_window_size: 10
+              size: 3
+              retriever:
+                knn:
+                  field: "textvector"
+                  query_vector: [ 0.5, 0.2, 0.4, 0.4 ]
+                  k: 10
+                  num_candidates: 10
+
+  - match: { hits.total.value: 8 }
+  - length: { hits.hits: 3 }
+  - match: { hits.hits.0._source.textbody: "second text" }
+  - match: { hits.hits.1._source.textbody: "third text" }
+  - match: { hits.hits.2._source.textbody: "sixth text" }
```

Comment:
Interesting that we returned different overall results here (sixth instead of first). I assume that's just the output of the formula, but it would be good to note this for explanation in a future blog post, go into why that may have happened.